app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.kt

```diff
@@ -0,0 +1,743 @@
+import nerd.tuxmobil.fahrplan.congress.net.ParseResult
+import nerd.tuxmobil.fahrplan.congress.net.ParseScheduleResult
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
+import nerd.tuxmobil.fahrplan.congress.repositories.OnSessionsChangeListener
+import nerd.tuxmobil.fahrplan.congress.repositories.SessionsTransformer
+import nerd.tuxmobil.fahrplan.congress.sharing.JsonSessionFormat
+import nerd.tuxmobil.fahrplan.congress.sharing.SessionSharer
+import nerd.tuxmobil.fahrplan.congress.sharing.SimpleSessionFormat
+import nerd.tuxmobil.fahrplan.congress.utils.FahrplanMisc
+import nerd.tuxmobil.fahrplan.congress.utils.Font
+import nerd.tuxmobil.fahrplan.congress.utils.TypefaceFactory
+import org.ligi.tracedroid.logging.Log
+import kotlin.collections.set
+
+class FahrplanFragment : Fragment(), SessionViewEventsHandler {
+
+    /**
+     * Interface definition for a callback to be invoked when a session view is clicked.
+     */
+    internal interface OnSessionClickListener {
+        /**
+         * Called when the session view has been clicked.
+         */
+        fun onSessionClick(sessionId: String)
+    }
+
+    companion object {
+
+        private const val LOG_TAG = "FahrplanFragment"
+        const val FRAGMENT_TAG = "schedule"
+        const val FAHRPLAN_FRAGMENT_REQUEST_CODE = 6166
+
+        private const val CONTEXT_MENU_ITEM_ID_FAVORITES = 0
+        private const val CONTEXT_MENU_ITEM_ID_SET_ALARM = 1
+        private const val CONTEXT_MENU_ITEM_ID_DELETE_ALARM = 2
+        private const val CONTEXT_MENU_ITEM_ID_ADD_TO_CALENDAR = 3
+        private const val CONTEXT_MENU_ITEM_ID_SHARE = 4
+        private const val CONTEXT_MENU_ITEM_ID_SHARE_TEXT = 5
+        private const val CONTEXT_MENU_ITEM_ID_SHARE_JSON = 6
+
+        const val FIFTEEN_MINUTES = 15
+        const val BOX_HEIGHT_MULTIPLIER = 3
+
+        private val sessionsTransformer = SessionsTransformer.createSessionsTransformer()
+
+    }
+
+    private lateinit var appRepository: AppRepository
+    private lateinit var inflater: LayoutInflater
+    private lateinit var alarmServices: AlarmServices
+    private lateinit var sessionViewDrawer: SessionViewDrawer
+    private lateinit var navigationMenuEntriesGenerator: NavigationMenuEntriesGenerator
+    private lateinit var errorMessageFactory: ErrorMessage.Factory
+    private lateinit var roomTitleTypeFace: Typeface
+    private lateinit var contextMenuView: View
+    private var conference: Conference? = null
+    private var scrollAmountCalculator: ScrollAmountCalculator? = null
+    private var onSessionClickListener: OnSessionClickListener? = null
+
+    private var mDay = 1
+    private var sessionId: String? = null
+    private var lastSelectedSession: Session? = null
+    private var scheduleData: ScheduleData? = null
+
+    private var displayDensityScale = 0f
+    private val adapterByRoomIndex: MutableMap<Int, SessionViewColumnAdapter> = HashMap()
```

Comment:
:broom: Change to `private val adapterByRoomIndex = mutableMapOf</* roomIndex */ Int, SessionViewColumnAdapter>()`

Reply:
:heavy_check_mark: Updated in f0f80685dd80301ff192b369d7e923fe6aa00819.